### ISSUE TITLE
Blank page: error vscode Handshake timed out 

### ISSUE THREAD

**Issue by hakanu** [coder/code-server#3027]
## OS/Web Information

- Web Browser: Google Chrome Version 89.0.4389.114 (Official Build) (64-bit)
- Local OS: Ubuntu 21.04
- Remote OS: Raspbian GNU/Linux 10 (buster)
- Remote Architecture: armv7l GNU/Linux
- `code-server --version`:3 .9.2

## Steps to Reproduce

1. SSH in to raspberry pi 
2. Run this `curl -fsSL https://code-server.dev/install.sh | sh`
3. Takes 22m usually
4. Go to hosted site through local network, it's blank and in the logs it gives  this error:
```
error vscode Handshake timed out {"token": "..."}
```
5. It was working fine before the upgrade.
6. Maybe you can also add some fresh install instructions. 

## Expected

Browser should show coder as usual

## Actual

Blank page.

## Screenshot

![Screenshot from 2021-04-04 02-11-43](https://user-images.githubusercontent.com/403473/113495693-2e4dad00-94eb-11eb-9fd6-a896b67e1aab.png)

## Notes

This issue can be reproduced in VS Code: Yes/No

**Comment by itsseanl** [comment]
Having this same issue on iPad safari. Tested my setup from a desktop and it worked fine there

**Comment by jsjoeio** [comment]
Hey @hakanu - sorry to hear you're having issues. Thanks for the report!

Can you verify if you experience this same issue using code-server 3.9.1?

Possibly related:
- https://github.com/cdr/code-server/issues/2975

**Comment by hakanu** [comment]
No it's with 3.9.2 the latest version as I mentioned in the first comment.

**Comment by jsjoeio** [comment]
@hakanu yes, I saw that. I'm wondering if you have a workaround for now :) Thanks for confirming.

**Comment by hakanu** [comment]
no worries :)

Unforunately not :/ It was working fine 2 versions before though. I should not have upgraded an already working system.

Is there a way to totally clean up the system and re-install?

I tried npm uninstall code-server and retry but no help. Any idea is welcome.

**Comment by jsjoeio** [comment]
Hmm..interesting. 

If you installed globally with npm, you should be able to do:
1. `npm uninstall -g code-server`
2. `npm install -g code-server@3.9.0` or which ever version was working before

Hope that helps!

**Comment by hakanu** [comment]
Thanks! same with 3.9.1 and 3.9.0 :/
I feel like something is wrong with the underlying vscode itself. Any suggestions like purging vscode from the system and then re-install?

**Comment by lokhmakov** [comment]
Same with install in Google Cloud Shell.

**Comment by jsjoeio** [comment]
> same with 3.9.1 and 3.9.0 :/

Argh. How about 3.8.1? 

> I feel like something is wrong with the underlying vscode itself

You may be able to go inside `code-server` and go inside the `lib/vscode` directory and run `npm rebuild`.  

It looks like someone had a similar issue in February - [see thread](https://github.com/cdr/code-server/discussions/2692)

**Comment by jkcoxson** [comment]
I am able to reproduce this on the newest build, the log is full of handshake errors and visual studio ends up with a blank screen.

**Comment by jkcoxson** [comment]
I can confirm this is affected on a clean install of 3.8.1. Is there some sort of dependency that broke it?

**Comment by jkcoxson** [comment]
No, I'm using x64, on Ubuntu

**Comment by jkcoxson** [comment]
Is there anything I can do to at least get something up and running? This is my main way to do my job at the moment, and something would be better than nothing. But if not, I'll be patient. Thanks for your work on this project.

**Comment by jsjoeio** [comment]
Hmm...3.9.1 _should_ work on Linux. 

Have you seen the [deploy code-server guides](https://github.com/cdr/deploy-code-server)? As a workaround, I would suggest using 3.9.1 on Railway. That should unblock you and help you get something working quickly. Sorry for blocking you and thank you for having patience + trying to help us figure this out!

**Comment by jkcoxson** [comment]
I was able to completely purge 3.9.2 with the help of Ben and reinstall using 3.9.1 and it worked for me. If it helps, I'm using Safari on my iPad, but it also didn't work on Chrome.

**Comment by jsjoeio** [comment]
Hi can you try upgrade to `3.9.3` and letting us know if you're still experiencing the same issue?

Feel free to post in the [release discussion](https://github.com/cdr/code-server/discussions/3075) as well.

**Comment by wingzero0** [comment]
@jsjoeio I get the same error in 3.9.2, 3.9.3, but not before 3.9.1 

Here is my running log from 3.9.3, I run it in docker, base image is ubuntu:20.04, docker host is also ubuntu:20.04.
but my docker host worked behind my company firewall, not sure whether if it is related.
```
[2021-04-09T04:16:02.521Z] info  code-server 3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1
[2021-04-09T04:16:02.525Z] info  Using user-data-dir ~/.local/share/code-server
[2021-04-09T04:16:02.543Z] info  Using config file ~/.config/code-server/config.yaml
[2021-04-09T04:16:02.543Z] info  HTTP server listening on http://0.0.0.0:9000
[2021-04-09T04:16:02.543Z] info    - Authentication is enabled
[2021-04-09T04:16:02.544Z] info      - Using password from ~/.config/code-server/config.yaml
[2021-04-09T04:16:02.544Z] info    - Not serving HTTPS
  ERR /root/.local/share/code-server/extensions [Error: ENOENT: no such file or directory, scandir '/root/.local/share/code-server/extensions'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: '/root/.local/share/code-server/extensions'
}
[2021-04-09T04:16:22.575Z] error Failed to get latest version {"error":"self signed certificate in certificate chain"}
Error: invalid distance too far back
    at Zlib.zlibOnError [as onerror] (zlib.js:182:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
[2021-04-09T04:16:22.605Z] warn  vscode invalid distance too far back
[2021-04-09T04:16:32.583Z] error vscode Handshake timed out {"token":"5f4a34e0-243b-4f9e-8b90-037747c124b5"}
  ERR /root/.local/share/code-server/extensions [Error: ENOENT: no such file or directory, scandir '/root/.local/share/code-server/extensions'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: '/root/.local/share/code-server/extensions'
}
[2021-04-09T04:16:46.315Z] error Failed to get latest version {"error":"self signed certificate in certificate chain"}
Error: invalid distance too far back
    at Zlib.zlibOnError [as onerror] (zlib.js:182:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
[2021-04-09T04:16:46.367Z] warn  vscode invalid distance too far back
[2021-04-09T04:16:56.344Z] error vscode Handshake timed out {"token":"4280cead-0a6f-4cbc-8485-5c2d5e43a63d"}
```


This is my Dockerfile without company CA cert. (sorry that I cannot attach the ca cert to reproduce the environment)
[https://github.com/wingzero0/codeserverUbuntu/blob/master/Dockerfile](https://github.com/wingzero0/codeserverUbuntu/blob/master/Dockerfile)

**Comment by wingzero0** [comment]
I made a docker-compose script for easier to describe my environment.

I pushed my docker image to docker hub [https://hub.docker.com/repository/docker/wingzzz2003/codeserver_ubuntu](https://hub.docker.com/repository/docker/wingzzz2003/codeserver_ubuntu)
tag 3.9.3 was built today and tag 3.9.1 was built 2 weeks ago

This is my docker-compose script [https://github.com/wingzero0/codeserverUbuntu/blob/master/docker-compose.yml](https://github.com/wingzero0/codeserverUbuntu/blob/master/docker-compose.yml)

3.9.3 showed 'error vscode Handshake timed out'
log from 3.9.3
```
[2021-04-09T05:18:16.016Z] info  code-server 3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1
[2021-04-09T05:18:16.017Z] info  Using user-data-dir ~/.local/share/code-server
[2021-04-09T05:18:16.032Z] info  Using config file ~/.config/code-server/config.yaml
[2021-04-09T05:18:16.032Z] info  HTTP server listening on http://0.0.0.0:8080
[2021-04-09T05:18:16.033Z] info    - Authentication is enabled
[2021-04-09T05:18:16.033Z] info      - Using password from ~/.config/code-server/config.yaml
[2021-04-09T05:18:16.033Z] info    - Not serving HTTPS
  ERR /root/.local/share/code-server/extensions [Error: ENOENT: no such file or directory, scandir '/root/.local/share/code-server/extensions'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: '/root/.local/share/code-server/extensions'
}
Error: invalid distance too far back
    at Zlib.zlibOnError [as onerror] (zlib.js:182:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
[2021-04-09T05:18:30.976Z] error Failed to get latest version {"error":"self signed certificate in certificate chain"}
[2021-04-09T05:18:30.979Z] warn  vscode invalid distance too far back
[2021-04-09T05:18:40.967Z] error vscode Handshake timed out {"token":"1d7b2efe-1532-4787-bc45-bad0f53c39a0"}
```

But 3.9.1 run fine. (to run 3.9.1, just simply changing docker-compose script to reference to tag 3.9.1)
log from 3.9.1
```
[2021-04-09T05:19:27.252Z] info  code-server 3.9.1 e0203f2a36c9b7036fefa50eec6cf8fa36c5c015
[2021-04-09T05:19:27.255Z] info  Using user-data-dir ~/.local/share/code-server
[2021-04-09T05:19:27.272Z] info  Using config file ~/.config/code-server/config.yaml
[2021-04-09T05:19:27.272Z] info  HTTP server listening on http://0.0.0.0:8080
[2021-04-09T05:19:27.273Z] info    - Authentication is enabled
[2021-04-09T05:19:27.273Z] info      - Using password from ~/.config/code-server/config.yaml
[2021-04-09T05:19:27.274Z] info    - Not serving HTTPS
[2021-04-09T05:19:53.710Z] error Failed to get latest version {"error":"self signed certificate in certificate chain"}
```

**Comment by jsjoeio** [comment]
> This is my docker-compose script

I like how simple/straightforward your docker-compose script is!

Thanks for the logs.

Bummer. Okay this is probably a separate issue then. We'll investigate.

**Comment by danyhoron** [comment]
I encountered the same issue in Chrome and Brave Browser on MacOS (on v3.9.2 and v3.9.3 but not on v3.9.1).
v3.9.3 works on Safari though.

**Comment by jsjoeio** [comment]
Hmm...I am on macOS and not able to reproduce with 3.9.3

![image](https://user-images.githubusercontent.com/3806031/114916211-62717980-9dd9-11eb-97b8-c5cb12cf8dd2.png)

@danyhoron do you see any logs in the browser or when running `code-server --log debug` with 3.9.3?

**Comment by danyhoron** [comment]
My instance runs in a docker container on another machine and I have an IIS (windows server) reverse proxy.
Here are the logs in Chrome:
![image](https://user-images.githubusercontent.com/9220101/114978371-cb1c2d00-9e91-11eb-9148-ac4e9ef165b6.png)

**Comment by danyhoron** [comment]
If I'm bypassing the reverse proxy and go straight to the docker instance it works in both Chrome and Brave.

Starting w/ v.3.9.2 it stopped working when using the reverse proxy.

**Comment by jsjoeio** [comment]
Ah, so maybe something broke in 3.9.3 with the reverse proxy. That at least gives us a place to start investigating. Thanks so much for the logs and screenshots @danyhoron! Super helpful 🙌

**Comment by hakanu** [comment]
My setup doesn't have any reverse proxy, locally it runs and but fails to handshake.

I'm suspicious about the raspberry pi's 32 bit ARM architecture. Because 3.9.2 works fine on my intel desktop on ubuntu 64 bit

**Comment by 12io** [comment]
> My setup doesn't have any reverse proxy, locally it runs and but fails to handshake.
> 
> I'm suspicious about the raspberry pi's 32 bit ARM architecture. Because 3.9.2 works fine on my intel desktop on ubuntu 64 bit

This could also be the case. I just installed it like in the past hour on a Pi 4 with armv7l after installing node12. So it took the npm way of installing it. Created a self signed cert due to testing and got the same error. But also there is a warning. I don't know how much this is helpful, but there is also a complaint about zlib. Here a short "full" log from "Starting code-server" to "visit the page":

```
pi@picroft:~ $ code-server
[2021-04-16T18:32:19.757Z] info  code-server 3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1
[2021-04-16T18:32:19.761Z] info  Using user-data-dir ~/.local/share/code-server
[2021-04-16T18:32:19.827Z] info  Using config file ~/.config/code-server/config.yaml
[2021-04-16T18:32:19.828Z] info  HTTP server listening on http://0.0.0.0:8443
[2021-04-16T18:32:19.828Z] info    - Authentication is disabled
[2021-04-16T18:32:19.828Z] info    - Using certificate for HTTPS: ~/mycroft-snakeoil.pem
Error: invalid distance too far back
    at Zlib.zlibOnError [as onerror] (zlib.js:182:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
[2021-04-16T18:32:43.965Z] warn  vscode invalid distance too far back
[2021-04-16T18:32:53.943Z] error vscode Handshake timed out {"token":"d56b2e87-8fb2-4d29-be42-f09857b8e683"}
Error: invalid distance too far back
    at Zlib.zlibOnError [as onerror] (zlib.js:182:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
[2021-04-16T18:32:54.396Z] warn  vscode invalid distance too far back
[2021-04-16T18:33:04.387Z] error vscode Handshake timed out {"token":"f8597587-f7df-40bd-b812-8b57f958dda4"}
```

**Comment by YaBoiMason** [comment]
I'm getting this on the latest ver. without any Zlib errors, just shows a blank screen and shows this:
```
Apr 18 12:36:36 neptune-2 systemd[1]: Starting code-server...
Apr 18 12:36:37 neptune-2 systemd[1]: Started code-server.
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.145Z] info  code-server 3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.149Z] info  Using user-data-dir ~/.local/share/code-server
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.225Z] info  Using config file ~/.config/code-server/config.yaml
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.226Z] info  HTTP server listening on http://0.0.0.0:8080
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.227Z] info    - Authentication is enabled
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.227Z] info      - Using password from ~/.config/code-server/config.yaml
Apr 18 12:36:39 neptune-2 code-server[32046]: [2021-04-18T16:36:39.228Z] info    - Using certificate for HTTPS: ~/.local/share/code-server/localhost.crt
Apr 18 12:37:03 neptune-2 code-server[32046]: [2021-04-18T16:37:03.539Z] error vscode Handshake timed out {"token":"7525f46e-987a-4acc-bb59-27c95bac7890"}
```
I've reinstalled twice and I've upgraded to Debian Unstable.

**Comment by owocean** [comment]
Latest version, installed today on Arch WSL 2
```
[2021-04-19T18:16:05.358Z] info  code-server 3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1
[2021-04-19T18:16:05.360Z] info  Using user-data-dir ~/.local/share/code-server
[2021-04-19T18:16:05.371Z] info  Using config file ~/.config/code-server/config.yaml
[2021-04-19T18:16:05.372Z] info  HTTP server listening on http://127.0.0.1:8080
[2021-04-19T18:16:05.372Z] info    - Authentication is enabled
[2021-04-19T18:16:05.372Z] info      - Using password from ~/.config/code-server/config.yaml
[2021-04-19T18:16:05.372Z] info    - Not serving HTTPS
[2021-04-19T18:16:20.747Z] error vscode Handshake timed out {"token":"2e70bc73-5d43-4227-9352-3417ef02d3d5"}
```
I've reinstalled multiple times and looked everywhere, not sure what is causing this problem. Same error as all of you, I just need to be able to do work.

**Comment by jsjoeio** [comment]
@owocean do you get the same error on 3.9.2 or 3.9.1?

And for alternative ways to get code-server running, take a look at the [deploy-code-server guide](https://github.com/cdr/deploy-code-server).

We're still investigating this issue in the meantime.

**Comment by jsjoeio** [comment]
> I get the same error on 3.9.3, 3.9.2, and 3.9.1

Thanks for confirming!

> I cannot afford alternative hosting

Completely understand. There are three free options - Railway, Heroku and Azure - offered in our guide. 

Keep an eye on the this issue in the meantime.

**Comment by code-asher** [comment]
We might be able to get some helpful information with `--log trace`. It could also be helpful to get the corresponding browser logs along with the server side ones.

**Comment by YaBoiMason** [comment]
I found out that if I connect to my server through it's local IP then I don't have any issues, but if I go to the domain name that I have pointed to my reverse proxy which does its magic and brings me to the code-server instance I get the blank screen. I can see the password screen just find, but the actual app is blank.

**Comment by YaBoiMason** [comment]
```
[2021-04-22T01:31:38.728Z] trace child:25744 got message {"message":{"type":"handshake","args":{"_":[],"bind-addr":"0.0.0.0:8080","auth":"password","password":"testpassword","cert":{"value":"/home/mason/.local/share/code-server/localhost.crt"},"config":"/home/mason/.config/code-server/config.yaml","log":"trace","user-data-dir":"/home/mason/.local/share/code-server","extensions-dir":"/home/mason/.local/share/code-server/extensions","verbose":true,"host":"0.0.0.0","port":8080,"cert-key":"/home/mason/.local/share/code-server/localhost.key","proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false}}}
[2021-04-22T01:31:38.735Z] info  code-server 3.9.3 fe2dc2deb08e378069891b622bb62ad1d261d1b1
[2021-04-22T01:31:38.736Z] info  Using user-data-dir ~/.local/share/code-server
[2021-04-22T01:31:38.736Z] trace Using extensions-dir ~/.local/share/code-server/extensions
[2021-04-22T01:31:38.795Z] info  Using config file ~/.config/code-server/config.yaml
[2021-04-22T01:31:38.796Z] info  HTTP server listening on http://0.0.0.0:8080
[2021-04-22T01:31:38.796Z] info    - Authentication is enabled
[2021-04-22T01:31:38.796Z] info      - Using password from ~/.config/code-server/config.yaml
[2021-04-22T01:31:38.797Z] info    - Using certificate for HTTPS: ~/.local/share/code-server/localhost.crt
[2021-04-22T01:31:46.272Z] trace heartbeat
[2021-04-22T01:31:46.286Z] debug forking vs code...
[2021-04-22T01:31:47.259Z] trace got message {"message":{"type":"ready"}}
[2021-04-22T01:31:47.260Z] debug setting up vs code...
[2021-04-22T01:31:47.265Z] debug vscode got message from code-server {"type":"init"}
[2021-04-22T01:31:47.269Z] trace vscode code-server message content {"message":{"type":"init","id":"XMeHFsvIbURxlkhcurD219W4","options":{"args":{"_":[],"bind-addr":"0.0.0.0:8080","auth":"password","password":"testpassword","cert":{"value":"/home/mason/.local/share/code-server/localhost.crt"},"config":"/home/mason/.config/code-server/config.yaml","log":"trace","user-data-dir":"/home/mason/.local/share/code-server","extensions-dir":"/home/mason/.local/share/code-server/extensions","verbose":true,"host":"0.0.0.0","port":8080,"cert-key":"/home/mason/.local/share/code-server/localhost.key","proxy-domain":[],"usingEnvPassword":false,"usingEnvHashedPassword":false},"remoteAuthority":"code.snwy.me"}}}
[2021-04-22T01:31:47.462Z] trace got message {"message":{"type":"options","id":"XMeHFsvIbURxlkhcurD219W4","options":{"workbenchWebConfiguration":{"remoteAuthority":"code.snwy.me","logLevel":0,"workspaceProvider":{"payload":[["userDataPath","/home/mason/.local/share/code-server"],["enableProposedApi","[]"]]}},"remoteUserDataUri":{"$mid":1,"path":"/home/mason/.local/share/code-server","scheme":"vscode-remote","authority":"code.snwy.me"},"productConfiguration":{"nameShort":"code-server","nameLong":"code-server","applicationName":"code-oss","dataFolderName":".vscode-oss","win32MutexName":"vscodeoss","licenseName":"MIT","licenseUrl":"https://github.com/microsoft/vscode/blob/main/LICENSE.txt","win32DirName":"Microsoft Code OSS","win32NameVersion":"Microsoft Code OSS","win32RegValueName":"CodeOSS","win32AppId":"{{E34003BB-9E10-4501-8C11-BE3FAA83F23F}","win32x64AppId":"{{D77B7E06-80BA-4137-BCF4-654B95CCEBC5}","win32arm64AppId":"{{D1ACE434-89C5-48D1-88D3-E2991DF85475}","win32UserAppId":"{{C6065F05-9603-4FC4-8101-B9781A25D88E}","win32x64UserAppId":"{{CC6B787D-37A0-49E8-AE24-8559A032BE0C}","win32arm64UserAppId":"{{3AEBF0C8-F733-4AD4-BADE-FDB816D53D7B}","win32AppUserModelId":"Microsoft.CodeOSS","win32ShellNameShort":"C&ode - OSS","darwinBundleIdentifier":"com.visualstudio.code.oss","linuxIconName":"com.visualstudio.code.oss","licenseFileName":"LICENSE.txt","reportIssueUrl":"https://github.com/cdr/code-server/issues/new","urlProtocol":"code-oss","extensionAllowedProposedApi":["ms-vscode.vscode-js-profile-flame","ms-vscode.vscode-js-profile-table","ms-vscode.github-browser","ms-vscode.github-richnav"],"builtInExtensions":[{"name":"ms-vscode.node-debug","version":"1.44.17","repo":"https://github.com/microsoft/vscode-node-debug","metadata":{"id":"b6ded8fb-a0a0-4c1c-acbd-ab2a3bc995a6","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.node-debug2","version":"1.42.5","repo":"https://github.com/microsoft/vscode-node-debug2","metadata":{"id":"36d19e17-7569-4841-a001-947eb18602b2","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.references-view","version":"0.0.77","repo":"https://github.com/microsoft/vscode-reference-view","metadata":{"id":"dc489f46-520d-4556-ae85-1f9eab3c412d","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.js-debug-companion","version":"1.0.8","repo":"https://github.com/microsoft/vscode-js-debug-companion","metadata":{"id":"99cb0b7f-7354-4278-b8da-6cc79972169d","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.js-debug","version":"1.54.5","repo":"https://github.com/microsoft/vscode-js-debug","metadata":{"id":"25629058-ddac-4e17-abba-74678e126c5d","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}},{"name":"ms-vscode.vscode-js-profile-table","version":"0.0.11","repo":"https://github.com/microsoft/vscode-js-profile-visualizer","metadata":{"id":"7e52b41b-71ad-457b-ab7e-0620f1fc4feb","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}}],"webBuiltInExtensions":[{"name":"ms-vscode.github-browser","version":"0.0.15","repo":"https://github.com/microsoft/vscode-github-browser","metadata":{"id":"c1bcff4b-4ecb-466e-b8f6-b02788b5fb5a","publisherId":{"publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherName":"ms-vscode","displayName":"Microsoft","flags":"verified"},"publisherDisplayName":"Microsoft"}}],"//":"https://github.com/VSCodium/vscodium/pull/155/files","documentationUrl":"https://go.microsoft.com/fwlink/?LinkID=533484#vscode","keyboardShortcutsUrlMac":"https://go.microsoft.com/fwlink/?linkid=832143","keyboardShortcutsUrlLinux":"https://go.microsoft.com/fwlink/?linkid=832144","keyboardShortcutsUrlWin":"https://go.microsoft.com/fwlink/?linkid=832145","introductoryVideosUrl":"https://go.microsoft.com/fwlink/?linkid=832146","tipsAndTricksUrl":"https://go.microsoft.com/fwlink/?linkid=852118","newsletterSignupUrl":"https://www.research.net/r/vsc-newsletter","commit":"fe2dc2deb08e378069891b622bb62ad1d261d1b1","date":"2021-04-08T18:29:22Z","version":"1.54.2","extensionsGallery":{"serviceUrl":"https://extensions.coder.com/api","itemUrl":"","controlUrl":"","recommendationsUrl":""}},"nlsConfiguration":{"locale":"en","availableLanguages":{}},"commit":"fe2dc2deb08e378069891b622bb62ad1d261d1b1"}}}
[2021-04-22T01:31:48.604Z] debug vscode got message from code-server {"type":"socket"}
[2021-04-22T01:31:48.605Z] trace vscode code-server message content {"message":{"type":"socket","query":{"reconnectionToken":"afb44490-fdb1-4a01-841e-4cf391f7045a","reconnection":"false","skipWebSocketFrames":"false"},"permessageDeflate":true}}
[2021-04-22T01:31:48.611Z] trace vscode Protocol handshake {"token":"afb44490-fdb1-4a01-841e-4cf391f7045a"}
[2021-04-22T01:31:58.615Z] error vscode Handshake timed out {"token":"afb44490-fdb1-4a01-841e-4cf391f7045a"}
```
Here's the output with `--log trace`.

**Comment by code-asher** [comment]
Ah OK this often happens because the reverse proxy isn't set up to proxy web sockets and so the socket connection fails. We might be able to pinpoint the issue if you can provide more details on the reverse proxy you're using and the configuration for it.

You could also use `--link` which will set up proxying for you through our cloud.

**Comment by YaBoiMason** [comment]
I'm using [frp](https://github.com/fatedier/frp), and my configuration is [here](https://gist.github.com/YaBoiMason/bfc46c6eab861711c417de1bcd631292).

**Comment by eng-dibo** [comment]
+1 ->  v3.9.1

**Comment by hakanu** [comment]
My problem is resolved with 3.9.3 on my raspberry pi over LAN (192.168.0.x address). Thanks.

**Comment by luojiansh** [comment]
v3.9.3

For me the timeout goes away when I open Chrome DevTools during every page reload.

**Comment by ChrisG661** [comment]
I am experiencing the same with v3.9.3.
I think it is something to do with the node version. Previously I have installed the node v10 from ubuntu repository and I have no problem with it, but then I removed node and reinstalled the v14 LTS version from nodesource because I need it to build a project. After I rebooted, the issue happens.

**Comment by TobiRoby** [comment]
I suddenly experienced the time out issue in my setup two days ago.
Here is how I solved it.

Setup:
- local: wsl ubuntu 20.04
- remote: code-server (3.9.3) in docker on ubuntu 18.04-LTS running in azure
- network VPN GlobalProtect

Besides the timeouts I also noticed that initializing a ssh connection took up to 10 seconds.
After a quick web search pointed me towards not optimal/ compatible network settings with VPN.
So I set the MTU settings of my local network adapter to 1200.
Not only is ssh blazing fast now, but all code-server timeouts vanished.

So the timeouts are related to the network and not code-server itself.
But I still have no clue, why my setup stopped working from one day to the other...

**Comment by facundomedica** [comment]
I got this error randomly and in the console I see: `{"error":"ENOENT: no such file or directory, open '/usr/lib/code-server/dist/pages/pages/vscode.js.map'"}` Note the double "/pages/pages". Version 3.9.3. It also happened with 3.11.0 but it's like solving itself randomly? I'm using the --link flag.
It solves itself randomly eventually 🤪

**Comment by jsjoeio** [comment]
That is very unusual. Thanks for the note @facundomedica!

**Comment by pyIronManCoder** [comment]
> I suddenly experienced the time out issue in my setup two days ago. Here is how I solved it.
> 
> Setup:
> 
> * local: wsl ubuntu 20.04
> * remote: code-server (3.9.3) in docker on ubuntu 18.04-LTS running in azure
> * network VPN GlobalProtect
> 
> Besides the timeouts I also noticed that initializing a ssh connection took up to 10 seconds. After a quick web search pointed me towards not optimal/ compatible network settings with VPN. So I set the MTU settings of my local network adapter to 1200. Not only is ssh blazing fast now, but all code-server timeouts vanished.
> 
> So the timeouts are related to the network and not code-server itself. But I still have no clue, why my setup stopped working from one day to the other...

Hello, could you please specify what error you recieved, I am getting "[2023-05-03T07:39:06.690Z] error timed out" when I run code-server command. I am running it on a local Raspberry Pi 0 W (Installed via "yarn global add code-server" command and it was a successful install). I check my MTU on my router it shows 1500. I tried on 3 wifi networks (different Access Points) still shows same error.
